openedx/features/colaraz_features/middlewares.py

```diff
@@ -93,13 +94,16 @@ def _get_current_url(self, request):
 
         return '{}://{}{}'.format(schema, domain, request.path)
         
-    def _get_user_site_domain(self, user, session):
+    def _get_user_site_identifier(self, user, session):
         """
         Returns domain of site associated with the User.
         """
-        if not session.get('user_site_domain', None):
-            session['user_site_domain'] = UserAttribute.get_user_attribute(user, 'created_on_site')
-        return session.get('user_site_domain', None)
+        is_new_user = False
+        if not session.get('user_site_identifier', None):
+            profile, is_new_user = ColarazUserProfile.objects.get_or_create(user=user)
```

Comment:
When the user is getting registered on open edX, its account doesn't have appropriate properties so in the first go have to create his colaraz profile and allow him to complete the registration process, but he will never enter that site which he doesn't belong too.